File: ocs_samples/library_samples/Java/ocs_sample_library_preview/src/main/java/com/github/osisoft/ocs_sample_library_preview/sds/SdsStreamExtrapolation.java
/** SdsStreamExtrapolation.java
 * 
 */

package  com.github.osisoft.ocs_sample_library_preview.sds;

/** 
 * SdsStreamExtrapolation 0-3
 */
public enum SdsStreamExtrapolation {

    All(0),
    None(1),
    Forward(2),
    Backward(3);

    private final int SdsStreamExtrapolation;

    private SdsStreamExtrapolation(int id) {
        this.SdsStreamExtrapolation = id;
    }

    /**
     * gets the value
     * @return
     */
    int getValue() {
        return SdsStreamExtrapolation;
    }
}

File: ocs_samples/library_samples/Python3/ocs_sample_library_preview/SDS/SdsStreamExtrapolation.py
# SdsStreamExtrapolation.py
#

from enum import Enum


class SdsStreamExtrapolation(Enum):
    """
    SdsStreamExtrapolation 0 -3
    """
    All = 0
    Nonex = 1
    Forward = 2
    Backward = 3
